src/cpp/server/ServerMain.cpp

```diff
@@ -571,7 +572,12 @@ int main(int argc, char * const argv[])
             return true;
          };
 
-         error = file_utils::changeOwnership(serverDataDir, options.serverUser(), true, shouldChown);
+         system::User serverUser;
```

Comment:
In what contexts might this fail? I'm mainly wondering because (IIUC) this adds a new way for `rserver` to fail to start and just want to make sure we understand the failure modes (and are convinced they're rare or understandable + easy to diagnose).

Alternatively: what do you think about using `options.serverUser()` by default, and having an error here be a warning rather than fatal? Or would this ultimately just fail somewhere else further down?

Reply:
The work to get the `system::User` and the error check for that were done inside `file_utils::changeOwnership` before. The SDK options class has a `Error error = serverUser(system::User& out_user)` method, so I wanted this function to take the User object rather than the username.

It _should_ only fail if the server user doesn't exist, which really shouldn't be possible at this point, since a misconfigured rserver.conf should have caused the process to fail much earlier than this.

Reply:
That makes sense. Thanks!